src/CMakeLists.txt

```
@@ -94,6 +94,7 @@ target_compile_definitions(
             $<$<BOOL:${DLAF_ASSERT_HEAVY_ENABLE}>:DLAF_ASSERT_HEAVY_ENABLE>
             DLAF_FUNCTION_NAME=$<IF:$<BOOL:is_pretty_function_available>,__PRETTY_FUNCTION__,__func__>
             $<$<BOOL:${DLAF_WITH_MKL}>:DLAF_WITH_MKL>
+            $<$<BOOL:${DLAF_WITH_MKL_LEGACY}>:DLAF_WITH_MKL>
```

This is not a typo, right? I.e. you didn't intend to do
```suggestion
            $<$<BOOL:${DLAF_WITH_MKL_LEGACY}>:DLAF_WITH_MKL_LEGACY>
```
? I'm ok with this if we don't need to be able to distinguish between the two (we might still be able to do it through other means).

---

This is not a typo, since we want to treat them the same way in CPP sources, but from the CMake point of view we want to keep separated the two logics between old and new one.